### PR TITLE
feat: Support multiple entry points with configurable HTML tags

### PR DESCRIPTION
**Pull request by pswies**
This PR enables the existence of multiple entry points instead of a single index.html.

No functionality is changed with this change. It creates, however, a possibility to maintain several static pages corresponding to various URL paths of the application which is beneficial for SEO.

Usage is described in the README.

### CODE REVIEW

**Comment by jaredvu** [comment]
Screen Shot 2024-03-27 at 8 53 16 AM

Preview build seems to be failing @pswies

**Comment by pswies** [comment]
@jaredvu I suspect that the `generate-entry-points.js` script isn't being run before the build starts. To ensure that the PR adds this change:

https://github.com/dydxprotocol/v4-web/pull/398/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R12

If you are overriding the build command in your Vercel configuration, could you please try running it with `pnpm run build:generate-entry-points && ` before `tsc && vite build` ?

**Comment by aforaleka** [comment]
updating the v4-staging deployment to use `pnpm run build` instead of `vite build` fixed the deployment preview! @jaredvu

File: vercel.json [hunk 1]
```diff
@@ -1,3 +1,9 @@
 {
-  "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
-}
+  "buildCommand": "pnpm run build:generate-entry-points && vite build --mode staging && pnpm run build:inject-app-deeplinks",
```
Comment (aforaleka) [inline vercel.json:2]: will adding this override the build command for our testnet environmet?

Comment (pswies) [reply vercel.json:2]: By all means. The last commit was deliberetly added as a temporary measure, just in this PR, to see if the build would work if the command was adjusted. This was discussed with Jared out of band. Let me remove this change now.

Comment (pswies) [reply vercel.json]: Commit removed.

**Review by aforaleka — APPROVED** [review]

**Pull request merged**